ginipaybank/src/main/java/net/gini/pay/bank/capture/util/SimpleBusEventStore.kt

```diff
@@ -0,0 +1,79 @@
+package net.gini.pay.bank.capture.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.edit
+import org.slf4j.LoggerFactory
+
+internal class SimpleBusEventStore(context: Context) {
+    private val mSharedPreferences: SharedPreferences =
+        context.getSharedPreferences(APP_SIMPLE_EVENTS, Context.MODE_PRIVATE)
+
+    private val appSimpleEventListeners = mutableSetOf<EventChangeListener>()
```

Comment:
Instead of managing listeners you should have used a flow. 

Flows have the advantage of being canceled when the coroutine scope gets canceled so you don't have to call an unregister.

Reply:
OK, great. I didn't know couroutines are added and I can use them.